executor/grant.go

```diff
@@ -487,6 +487,13 @@ func (e *GrantExec) grantDBLevel(priv *ast.PrivElem, user *ast.UserSpec, interna
 		dbName = e.ctx.GetSessionVars().CurrentDB
 	}
 
+	// Attempts to use GRANT ALL as shorthand for granting privileges
+	// at the database leval fail with an error
+	// See https://dev.mysql.com/doc/refman/8.0/en/performance-schema-table-characteristics.html for more detail
+	if strings.EqualFold(dbName, "performance_schema") && priv.Priv == mysql.AllPriv {
```

Comment:
It's only denied `grant all` statement, TiDB also has different behavior with MySQL in `grant alter`, `grant create`, `grant execute` and so on.

Reply:
Maybe the best way to fix this issue is to add a permission check for every table?

Reply:
IMO we can extend the `Priv` check range here to have the same behavior with MySQL, but I wonder why the privilege management works incompatible on `performance_schema` and whether we can fix it via working on that as your suggestion, instead of introducing a special case like this PR. (I am not so familiar to TiDB codebase though.....) cc. @morgo

Reply:
- we can extend the Priv check range here to have the same behavior with MySQL  - Yes
- why the privilege management works incompatible on `performance_schema`  - `performance_schema` is a special system database in MySQL and TiDB ignores the privilege management part previously
- whether we can fix it via working on that as your suggestion - We can try granting every privilege to `performance_schema.*` in MySQL to see the behavior and then copy it to TiDB. Besides, MySQL also forbids granting some privileges to one single table in `performance_schema`.

Reply:
Got it. I will try granting privileges in MySQL and copy the results to TiDB. I think in this PR we can mainly focus on database level privileges. Thank you for your review~